packages/block-editor/src/components/roving-tab-index/item.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, Children, cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useRovingTabIndexContext } from './context';
+
+export default function RovingTabIndexItem( { children } ) {
+	const ref = useRef();
+	const lastFocusedElement = useRovingTabIndexContext();
+	let tabIndex;
+
+	if ( lastFocusedElement ) {
+		tabIndex = lastFocusedElement === ref.current ? 0 : -1;
+	}
+
+	// Ensure a single child.
+	Children.only( children );
+
+	// Override the tabIndex and ref props to the child element.
+	return cloneElement( children, {
+		tabIndex,
+		ref,
```

Comment:
`cloneElement` is pretty magical. My concern is that in usage, it's not particularly clear what the component does.

Alternatively a render callback is an option:
```
<RovingTabIndex.Item>
  ( tabIndex, ref ) => <Button ref={ ref } tabIndex={ tabIndex } />
</RovingTabIndex.Item>
```
It's a bit more code, but at least makes it clear that `ref` and `tabIndex` have to be handled by the child.